tuf/signed/verify.go

```diff
@@ -107,3 +108,18 @@ func VerifySignature(msg []byte, sig *data.Signature, pk data.PublicKey) error {
 	sig.IsValid = true
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks if the private key and the public keys forms valid key pairs.
+// This should work with both ecdsa-x509 certificate PublicKey as well as ecdsa PublicKey
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
+	pubKeyID, err := utils.CanonicalKeyID(pubKey)
+	privKeyID := privKey.ID()
```

Comment:
@cyli good catch, I missed this!

I agree that for this PR, at minimum we should check the error (`if err != nil {`) before any operations with the private key, and then check as @cyli suggests by doing `if privKey == nil || pubKeyID != privKey.ID {...`

Fixing `utils.CanonicalKeyID` would be a nice bonus if you're comfortable but agree that we can handle it in a separate PR 👍